internal/schema/schemanode.go

```diff
@@ -396,7 +393,7 @@ func schemaElementEquals(s, v parquet.SchemaElement) bool {
 		equals(s.Scale, v.Scale) &&
 		equals(s.Precision, v.Precision) &&
 		equals(s.FieldID, v.FieldID) &&
-		equals(s.LogicalType, v.LogicalType)
+		s.LogicalType.Equals(v.LogicalType)
```

Comment:
This has `nil` checks built in

Reply:
This is neat, I didn't realize that there is such a function already 😭

Reply:
> This has nil checks built in

If LogicalType is nil ...

Reply:
> > This has nil checks built in
> 
> If LogicalType is nil ...

So it should still be good. I had to check my Go fundamentals again, but as long as the `nil` is typed, the runtime takes care of it. Here's the code under the hood:

```go
func (p *LogicalType) Equals(other *LogicalType) bool {
	if p == other {
		return true
	} else if p == nil || other == nil {
		return false
	}
```

Ok good, so it catches nils. So i didnt want to commit an extra nil check for no reason. Here's a playground link i did to check myself: https://go.dev/play/p/_HuAyiIB4Hy so i think we're good here?